examples/web/nextjs/components/useApp.js

```diff
@@ -0,0 +1,15 @@
+// :snippet-start: use-app-hook
+import { useEffect, useState } from "react";
+import * as Realm from "realm-web";
+
+function useApp() {
+  const [app, setApp] = useState(null);
+  // Run in useEffect so that App is not created in server-side environment
+  useEffect(() => {
+    setApp(Realm.getApp(process.env.NEXT_PUBLIC_APP_ID));
+  }, []);
+  return app;
```

Comment:
I don't know Next.js good enough. But could this, simpler implementation work too?

```suggestion
  // Run in useMemo so that App is not created in server-side environment
  return useMemo(() => Realm.getApp(process.env.NEXT_PUBLIC_APP_ID));
```

Reply:
that would run serverside when it's not supposed to. 

wrapping w useEffect makes it only run client side.